### PR TITLE
Add compact map for shared sequences, single and maybe

### PR DESCRIPTION
**Pull request by gringoireDM**
This Pull request adds `compactMap` operators for `Single`, `SharedSequences`, and `Maybe`.

Since `compactMap` is now part of `ObservableType` I think that like other operators it should be included in the implementation of other Traits.

### CODE REVIEW

**Comment by NKorchagin** [comment]
👍

File: RxSwift/Traits/Single.swift [hunk 1]
```diff
@@ -213,6 +213,18 @@ extension PrimitiveSequenceType where Trait == SingleTrait {
             return Single(raw: self.primitiveSequence.source.map(transform))
     }
     
+    /**
```
Comment (kzaher) [inline RxSwift/Traits/Single.swift]: We can not add `compactMap` to `Single` because it breaks the contact.

Comment (gringoireDM) [reply RxSwift/Traits/Single.swift]: That's why it returns a `Maybe`, just like `filter` does.
Don't you think it's right?

**Comment by acecilia** [comment]
Can we also add it to Driver?

**Comment by NKorchagin** [comment]
> @gringoireDM Can you also add it to Driver?

It's already added via `SharedSequenceConvertibleType`

**Comment by gringoireDM** [comment]
@kzaher unit tests added

File: RxSwift/Traits/Single.swift (hunk 1 above)
Comment (kzaher) [reply RxSwift/Traits/Single.swift]: Oh, that's fine. I've misread that you are returning `Single`. My bad.

**Pull request merged**